x/nftmarket/keeper/nft_listing.go

```diff
@@ -58,6 +59,53 @@ func getTimeKey(prefix string, timestamp time.Time) []byte {
 	return bz
 }
 
+func (k Keeper) SaveNftListing(ctx sdk.Context, listing types.NftListing) {
+	k.SetNftListing(ctx, listing)
+	k.UpdateClass(ctx, listing)
+}
+
+func (k Keeper) UpdateClass(ctx sdk.Context, listing types.NftListing) {
+	switch listing.State {
+	case types.ListingState_LISTING:
+		k.SetClass(ctx, listing)
+	case types.ListingState_SUCCESSFUL_BID:
+		// todo delete nftid from class
+	}
+}
+
+func (k Keeper) SetClass(ctx sdk.Context, listing types.NftListing) {
+	store := ctx.KVStore(k.storeKey)
+	bzIdlist := store.Get(types.ClassKey(listing.ClassIdBytes()))
+	if bzIdlist == nil {
+		bz := k.cdc.MustMarshal(
+			&types.Class{
+				ClassId: listing.NftId.ClassId,
+				NftIds:  []string{listing.NftId.NftId},
+			},
+		)
+		store.Set(types.ClassKey(listing.ClassIdBytes()), bz)
+	} else {
+
+		// todo delete dumplicate nftid
+		class := types.Class{}
+		k.cdc.MustUnmarshal(bzIdlist, &class)
+		class.NftIds = append(class.NftIds, listing.NftId.NftId)
```

Comment:
It's pretty not common to have array of nfts and on the storage and it will continuously increase gas for the operation,

Reply:
Here is the data structure I added for storage
```
{
  "classId": {
    "classId":"string",
    "nftIds":["number"]
  }
}
```
Stored for each ClassId.
Its data is not large.
We need a Query to know which NFT was exhibited for each Class.